sycl/source/handler.cpp

```diff
@@ -73,6 +66,70 @@ handler::getOrInsertHandlerKernelBundle(bool Insert) const {
   return KernelBundleImpPtr;
 }
 
+// Returns a shared_ptr to kernel_bundle stored in the extended members vector.
+// If there is no kernel_bundle created:
+// returns newly created kernel_bundle if Insert is true
+// returns shared_ptr(nullptr) if Insert is false
+std::shared_ptr<detail::kernel_bundle_impl>
+handler::getOrInsertHandlerKernelBundle(bool Insert) const {
+
+  std::lock_guard<std::mutex> Lock(
+      detail::GlobalHandler::instance().getHandlerExtendedMembersMutex());
+
+  assert(!MSharedPtrStorage.empty());
+
+  std::shared_ptr<std::vector<detail::ExtendedMemberT>> ExendedMembersVec =
+      detail::convertToExtendedMembers(MSharedPtrStorage[0]);
+
+  return getOrInsertHandlerKernelBundleCommon(ExendedMembersVec, MQueue,
+                                              Insert);
+}
+
+// This function exhibits the same behavior as getOrInsertHandlerKernelBundle
+// but throws an exception with errc::invalid if a kernel bundle has been set
+// in the command group by a call to use_kernel_bundle.
+// If MarkSpecConstSet is true the command group is flagged as having had set
+// a specialization constant.
+std::shared_ptr<detail::kernel_bundle_impl>
+handler::getOrInsertNonExplicitHandlerKernelBundle(
+    bool Insert, bool MarkSpecConstSet) const {
+
+  std::lock_guard<std::mutex> Lock(
+      detail::GlobalHandler::instance().getHandlerExtendedMembersMutex());
+
+  assert(!MSharedPtrStorage.empty());
+
+  std::shared_ptr<std::vector<detail::ExtendedMemberT>> ExendedMembersVec =
```

Comment:
Discussed offline. Agreed to implement a `handler_impl` class which would constructed in handler constructor, stored in ExtendedMembers and contain all members needed.

Reply:
I have done as mentioned above. Sadly I don't believe we can move the old extended members at the moment as they are expected in cg.hpp.

Reply:
Thanks.